### PR TITLE
Add documentation for local file connector

### PR DESCRIPTION
**Pull request by nileema**

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+====================
+
+The local file connector allows querying data stored on the local
+filesystem of each worker.
+
+Configuration
+-------------
+
+To configure the local file connector, create a catalog properties file
+``etc/catalog/localfile.properties`` with the following contents:
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: This makes it appear as if the file *has* to be named that way. I'd just say something like "create a catalog properties file under etc/catalog (e.g., localfile.properties)"

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+.. code-block:: none
+
+    connector.name=localfile
+
+Using the Local File Connector
+------------------------------
+List the available system schemas::
+
+    SHOW SCHEMAS FROM system;
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: Copy & paste error? Why "system"?

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+This table contains the http request logs from each node on the cluster.
+
+Configuration Properties
+------------------------
+
+================================================== ============================================================
+Property Name                                      Description
+================================================== ============================================================
+
+    ``presto-logs.http-request-log-location``       Directory or file where http request logs are written
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: Is the difference between "presto-logs" and "presto-log" intentional?

File: presto-docs/src/main/sphinx/connector/localfile.rst (hunk 1 above)
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: This is the language we use elsewhere. It is simpler to tell people exactly what to do, then explain in detail elsewhere for advanced users.

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
+====================
+Local File Connector
+====================
+
+The local file connector allows querying data stored on the local
+filesystem of each worker.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: "file system" is two words

File: presto-docs/src/main/sphinx/connector/localfile.rst (hunk 1 above)
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: Or if you really care, copy the language from the MySQL connector.

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+To configure the local file connector, create a catalog properties file
+``etc/catalog/localfile.properties`` with the following contents:
+
+.. code-block:: none
+
+    connector.name=localfile
+
+Using the Local File Connector
+------------------------------
+List the available system schemas::
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: Newline after section headings

Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: I'm not sure this section is useful at all. Does the connector even have multiple schemas?

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    SHOW SCHEMAS FROM system;
+
+
+Local File Connector Schemas and Tables
+---------------------------------------
+The local file connector provides a single schema ``logs`` that contains tables for logs.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: Newline

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+You can see all the available tables by running ``SHOW TABLES``::
+
+        SHOW TABLES FROM localfile.logs;
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: This should be indented four space

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: Remove extra blank line

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+``http_request_log``
+^^^^^^^^^^^^^^^^^^^^
+This table contains the http request logs from each node on the cluster.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: Newline

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+Configuration Properties
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: Move this section above, after the "Configuration" section

File: presto-docs/src/main/sphinx/connector/localfile.rst (hunk 3 above)
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: This should be left aligned, no indentation

Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: Capitalize "HTTP"

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+------------------------
+
+================================================== ============================================================
+Property Name                                      Description
+================================================== ============================================================
+
+    ``presto-logs.http-request-log-location``       Directory or file where http request logs are written
+    ``presto-log.http-request-log-pattern``         If http-request-log-location is a directory, file name regex for files in this directory
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: Use formatting for `http-request-log-location`. Or say "If the log location is a directory, ..."

Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: This description is a bit confusing. How about something like "If the log location is a directory, this regex is used to match file names in the directory"

Comment (electrum) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: Also, is it really a regex? If so, please update the connector to use globbing, as that is standard for file matching:

https://docs.oracle.com/javase/tutorial/essential/io/fileOps.html#glob

File: presto-docs/src/main/sphinx/connector/localfile.rst [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+---------------------------------------
+
+The local file connector provides a single schema ``logs`` that contains tables for logs.
+You can see all the available tables by running ``SHOW TABLES``::
+
+    SHOW TABLES FROM localfile.logs;
+
+``http_request_log``
+^^^^^^^^^^^^^^^^^^^^
+This table contains the http request logs from each node on the cluster.
```
Comment (martint) [inline presto-docs/src/main/sphinx/connector/localfile.rst]: capitalize HTTP

File: presto-local-file/src/main/java/com/facebook/presto/localfile/DataLocation.java [hunk 13]
```diff
@@ -80,13 +81,21 @@ public File getLocation()
...
+                builder.add(path.toFile());
+            }
+            List<File> files = builder.build();
 
-        Arrays.sort(files, (o1, o2) -> Long.compare(o2.lastModified(), o1.lastModified()));
-        return ImmutableList.copyOf(files);
+            if (files.isEmpty()) {
+                throw new PrestoException(LOCAL_FILE_ERROR_CODE, format("Failed to list files at %s", location));
+            }
+            return files.stream().sorted((o1, o2) -> Long.compare(o2.lastModified(), o1.lastModified())).collect(Collectors.toList());
```
Comment (martint) [inline presto-local-file/src/main/java/com/facebook/presto/localfile/DataLocation.java]: split into multiple lines

File: presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileConfig.java [hunk 14]
```diff
@@ -39,8 +39,8 @@ public String getHttpRequestLogFileNamePattern()
         return httpRequestLogFileNamePattern;
     }
 
-    @Config("presto-log.http-request-log-pattern")
-    @ConfigDescription("If http-request-log-location is a directory, file name regex for files in this directory")
+    @Config("presto-logs.http-request-log-pattern")
```
Comment (martint) [inline presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileConfig.java]: I'd rename the option to `presto-logs.http-request-log.pattern`

**Comment by martint** [comment]
Other than minor comments, it looks good.

**Pull request merged**